homework/index.js

```diff
@@ -30,14 +30,105 @@
     return elem;
   }
 
+  function renderError(error) {
+    const root = document.getElementById('root');
+    createAndAppend('h1', root, { text: error.message });
+  }
+
+  function createOptions(repositoryDetails, select) {
+    repositoryDetails.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: index,
+      });
+    });
+  }
+
+  function addRow(tbody, labelText, value, str) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${labelText}`, class: str });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepoInformation(repositories, container) {
+    container.innerHTML = ' ';
+    const table = createAndAppend('table', container, { class: 'infoTable' });
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository: ', '', 'explanations');
+    createAndAppend('a', firstRow.lastChild, {
+      href: `${repositories.html_url}`,
+      target: 'blank',
+      text: `${repositories.name}`,
+    });
+    addRow(tbody, 'Description:', repositories.description, 'explanations');
+    addRow(tbody, 'Fork:', repositories.forks, 'explanations');
+    addRow(tbody, 'Updated:', `${repositories.updated_at}`.substring(0, 10), 'explanations');
+  }
+
+  function renderContributorsInformation(repositories, container) {
```

Comment:
This function gets a single `repository` as its first argument. Your name `repositories` (implying plural or, in other words, an array) is misleading, to say the least or, more bluntly stated, completely incorrect.